# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PreallocatedProcessManager.cpp
# Commit: 80193c01e94b
# Full Hash: 80193c01e94bbd93990445cacc7943af1b935cba
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2024-06-25 09:36:02
# Regressor Bug: 1728331
# File Overlap Count: 1
# Description:
#   Bug 1728331 - Part 1: Remove e10s process recycling, r=smaug
#   
#   This logic did nothing in Fission, however it did have an effect for
#   e10s-Android, which will no longer have process recycling. I expect the impact
#   of this to be minimal given Android's tendency to kill processes already.
# ==============================================================================

diff -r a8a15b242d9d -r 80193c01e94b dom/ipc/PreallocatedProcessManager.cpp
--- a/dom/ipc/PreallocatedProcessManager.cpp	Mon Jun 24 23:19:27 2024 +0000
+++ b/dom/ipc/PreallocatedProcessManager.cpp	Mon Jun 24 23:19:27 2024 +0000
@@ -371,14 +371,6 @@
     process->ShutDownProcess(ContentParent::SEND_SHUTDOWN_MESSAGE);
     // drop ref and let it free
   }
-
-  // Make sure to also clear out the recycled E10S process cache, as it's also
-  // controlled by the same preference, and can be cleaned up due to memory
-  // pressure.
-  if (RefPtr<ContentParent> recycled =
-          ContentParent::sRecycledE10SProcess.forget()) {
-    recycled->MaybeBeginShutDown();
-  }
 }
 
 inline PreallocatedProcessManagerImpl*
